library/core/src/hint.rs

```diff
@@ -45,6 +45,7 @@ use crate::intrinsics;
 #[inline]
 #[stable(feature = "unreachable", since = "1.27.0")]
 #[rustc_const_unstable(feature = "const_unreachable_unchecked", issue = "53188")]
+#[track_caller]
```

Comment:
Why this?

Reply:
Ah I guess that's due to the call in `const_panic_fmt`.

However, if we make `track_caller` only relevant for panic errors, then this will have no effect. And it should not happen anyway. So IMO we shouldn't unnecessarily complicate codegen of this function.

Reply:
Oh, yes. This was left over from me playing a bit with what happens in case the unreachable branch is hit. Removing it.

Reply:
Though it does seem to improve the output in case it happens:

Output of `:src/test/ui/consts/const_unsafe_unreachable_ub.rs`:

Before:
```
error[E0080]: evaluation of constant value failed
  --> $SRC_DIR/core/src/hint.rs:LL:COL
   |
LL |     unsafe { intrinsics::unreachable() }
   |              ^^^^^^^^^^^^^^^^^^^^^^^^^
   |              |
   |              entering unreachable code
   |              inside `unreachable_unchecked` at $SRC_DIR/core/src/hint.rs:LL:COL
   | 
  ::: $DIR/const_unsafe_unreachable_ub.rs:7:18
   |
LL |         false => std::hint::unreachable_unchecked(),
   |                  ---------------------------------- inside `foo` at $DIR/const_unsafe_unreachable_ub.rs:7:18
...
LL | const BAR: bool = unsafe { foo(false) };
   |                            ---------- inside `BAR` at $DIR/const_unsafe_unreachable_ub.rs:11:28
```

After:
```
error[E0080]: evaluation of constant value failed
  --> $DIR/const_unsafe_unreachable_ub.rs:7:18
   |
LL |         false => std::hint::unreachable_unchecked(),
   |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |                  |
   |                  entering unreachable code
   |                  inside `foo` at $DIR/const_unsafe_unreachable_ub.rs:7:18
...
LL | const BAR: bool = unsafe { foo(false) };
   |                            ---------- inside `BAR` at $DIR/const_unsafe_unreachable_ub.rs:11:28
```

But that's unrelated to this PR. So removing it from this one.